### PR TITLE
Feature/jwt improvements

### PR DESCRIPTION
**Pull request by italolelis**
## What does this PR do?

- Improves the JWT on Janus admin API
- Adds some useful commands for janus cli

### CODE REVIEW

File: docs/quick_start/README.md [hunk 1]
```diff
@@ -7,11 +7,11 @@
 
 ## Download
 
-You can get Janus for nearly any OS and architecture. You can get the latest Janus realease on [Github](https://github.com/hellofresh/janus/releases).
+You can get Janus for nearly any OS and architecture. You can get the latest Janus release on [Github](https://github.com/hellofresh/janus/releases).
 
 ## Install and run
 
-We higly recommend you to use one of your examples to start. Let's see the [front-proxy](/examples/front-proxy) example:
+We highly recommend you to use one of your examples to start. Let's see the [front-proxy](/examples/front-proxy) example:
```
Comment (rafaeljesus) [inline docs/quick_start/README.md]: `of our examples`

File: docs/quick_start/authenticating.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+## Authenticating
+
+To start using the Janus adminstration API you need to get a [JSON Web Token](https://jwt.io) and provide it in every single request using the `Authorization` header.
```
Comment (rafaeljesus) [inline docs/quick_start/authenticating.md]: `administration`

Comment (rafaeljesus) [reply docs/quick_start/authenticating.md]: dunno, but maybe `you need to issue a new token`

File: features/bootstrap/context_request.go [hunk 3]
```diff
@@ -199,12 +200,12 @@ func (c *requestContext) requestJWTTokenIsNotSet() error {
 
 func (c *requestContext) requestJWTTokenIsValidAdminToken() error {
 	jwtConfig := jwt.NewGuard(c.adminCred)
-	accessToken, err := jwt.IssueAdminToken(jwtConfig.SigningMethod, make(map[string]interface{}), jwtConfig.Timeout)
+	accessToken, err := jwt.IssueAdminToken(jwtConfig.SigningMethod, jwtgo.MapClaims{}, jwtConfig.Timeout)
 	if nil != err {
 		return fmt.Errorf("Failed to issue JWT: %v", err)
```
Comment (rafaeljesus) [inline features/bootstrap/context_request.go:205]: `errors.Wrap(err, "Failed to issue JWT")`

File: pkg/jwt/handler.go [hunk 4]
```diff
@@ -95,7 +102,31 @@ func (j *Handler) Refresh() http.HandlerFunc {
...
+
+func extractAccessToken(r *http.Request) (string, error) {
+	// We're using OAuth, start checking for access keys
+	authHeaderValue := r.Header.Get("Authorization")
+	parts := strings.Split(authHeaderValue, " ")
+	if len(parts) < 2 {
+		return "", errors.New("attempted access with malformed header, no auth header found")
+	}
+
+	if strings.ToLower(parts[0]) != "bearer" {
```
Comment (rafaeljesus) [inline pkg/jwt/handler.go]: extract a const for this `strings.ToLower(parts[0]) != bearer`

File: features/bootstrap/context_request.go (hunk 3 above)
Comment (italolelis) [reply features/bootstrap/context_request.go:205]: this are tests... I'm not willing to change it to use a library for errors

Comment (rafaeljesus) [reply features/bootstrap/context_request.go:205]: Ah indeed, I didn't notice it was tests

**Comment by rafaeljesus** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/193/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway) [Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/193/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway)

**Pull request merged**